scripts/docs/snippetadapter.js

```diff
@@ -7,83 +7,244 @@
+				const snippetBasePath = snippetData.snippetSources.js.replace( snippetBasePathRegExp, '' );
+
+				const dependencySnippet = {
+					snippetSources: getSnippetSourcePaths( snippetBasePath, dependencyName ),
+					snippetName: dependencyName,
+					outputPath: snippetData.outputPath,
+					destinationPath: snippetData.destinationPath,
+					requiredFor: snippetData
+				};
+
+				if ( !dependencySnippet.snippetSources.js ) {
+					throw new Error( `Missing snippet source for "${ dependencySnippet.snippetName }".` );
+				}
+
+				dependencySnippet.snippetConfig = readSnippetConfig( dependencySnippet.snippetSources.js );
+				dependencySnippet.snippetConfig.language = dependencySnippet.snippetConfig.language || DEFAULT_LANGUAGE;
+
+				snippetsDependencies.set( dependencyName, dependencySnippet );
+			}
+		}
 	}
 
-	const snippetConfig = readSnippetConfig( data.snippetSource.js );
-	const outputPath = path.join( data.outputPath, data.snippetPath );
+	// Add all dependencies to the snippet collection.
+	for ( const snippetData of snippetsDependencies.values() ) {
+		snippets.add( snippetData );
+	}
 
-	const webpackConfig = getWebpackConfig( {
-		entry: data.snippetSource.js,
-		outputPath,
-		language: snippetConfig.language,
-		production: data.options.production,
-		definitions: data.options.definitions || {}
-	} );
+	// Remove snippets that do not match to patterns specified in `options.whitelistedSnippets`.
+	if ( options.whitelistedSnippets ) {
+		filterWhitelistedSnippets( snippets, options.whitelistedSnippets );
+	}
+
+	console.log( 'Number of snippets that will be built:', snippets.size );
+
+	const groupedSnippetsByLanguage = {};
+
+	// Group snippets by language. There is no way to build different languages in a single Webpack process.
+	// Webpack must be called as many times as different languages are being used in snippets.
+	for ( const snippetData of snippets ) {
+		if ( !groupedSnippetsByLanguage[ snippetData.snippetConfig.language ] ) {
+			groupedSnippetsByLanguage[ snippetData.snippetConfig.language ] = new Set();
+		}
+
+		groupedSnippetsByLanguage[ snippetData.snippetConfig.language ].add( snippetData );
+	}
+
+	// For each language prepare own Webpack configuration.
+	const webpackConfigs = Object.keys( groupedSnippetsByLanguage )
+		.map( language => {
+			return getWebpackConfig( groupedSnippetsByLanguage[ language ], {
+				language,
+				production: options.production,
+				definitions: options.definitions || {}
+			} );
+		} );
 
-	let promise;
+	let promise = Promise.resolve();
 
-	// See #530.
-	if ( webpackProcesses.has( outputPath ) ) {
-		promise = webpackProcesses.get( outputPath );
-	} else {
-		promise = runWebpack( webpackConfig );
-		webpackProcesses.set( outputPath, promise );
+	// Nothing to build.
+	if ( !webpackConfigs.length ) {
+		return promise;
+	}
+
+	for ( const config of webpackConfigs ) {
+		promise = promise.then( () => runWebpack( config ) );
```

Comment:
I'd check the `Promise.all()` too, but that's fine too.

Reply:
`Promise.all()` = 2 webpacks at the same time = (most probably) memory errors.